kite-data/kite-data-core/src/main/java/org/kitesdk/data/Datasets.java

```diff
@@ -225,14 +336,38 @@
   /**
    * Update a {@link Dataset} for the given dataset or view URI.
    * <p>
-   * URI formats are defined by {@code Dataset} implementations, but must begin
-   * with "dataset:" or "view:".
+   * You can add columns, remove columns, or change the data type of columns 
+   * in your dataset, provided you don't attempt a change that would corrupt 
+   * the data. Kite follows the guidelines in the Avro schema. See 
+   * <a href="http://kitesdk.org/docs/current/Schema-Evolution.html">Schema
+   * Evolution</a>.
+   * <p>
```

Comment:
This method updates the dataset's descriptor, not just the dataset's schema. This is how you would add or change properties.

This should link to the DatasetDescriptor.Builder's copy constructor. That allows you to start building a descriptor from an  existing one, which is the recommended way to build descriptors to pass into this method.

Also note that the dataset's format and partitioning can't be changed.